go.mod

```diff
@@ -1,11 +1,25 @@
 module github.com/spiffe/spire-plugin-sdk
 
-go 1.14
+go 1.18
```

Comment:
Hmm, I think we need to be careful about bumping this. Like go-spiffe, these SDKs are intended to be consumed by organizations that may or may not be on the latest release of Go, and some which are likely pinned to older versions for various reasons....

Reply:
Yeah that's a good point. I narrowed this PR to just include the dep updates and leave out the Go version bump.